Gemfile

```diff
@@ -16,6 +16,8 @@ branch = File.read(File.expand_path("../maintenance-branch", __FILE__)).chomp
   end
 end
 
+gem "rspec-support", :git => "https://github.com/nevinera/rspec-support.git", :branch => "nev--override-stderr-splitter-clone"
```

Comment:
This is not needed anymore when https://github.com/rspec/rspec-support/pull/598 is merged?

Reply:
I think that's correct, but I'll need a few minutes to refresh; I'll take a look later today. :-)

Reply:
Ah, so the resolution on the other PR was just to declare that the behavior would remain incorrect on jruby for now, and skip the relevant tests there? Should I update this PR to match? I didn't feel comfortable skipping support for such a significant ruby, but I admit I didn't make any progress on it, and haven't found time to come back to the problem in a while now :-\

Reply:
I am having a surprising amount of trouble successfully skipping these tests for the right containers. It sure seems like "JRuby 1.7 1.8 mode" has `RSpec::Support::Ruby.jruby?` returning true; do you know what's special about that setup? (I'll poke around again this evening)

Reply:
> "JRuby 1.7 1.8 mode" has RSpec::Support::Ruby.jruby? returning true

I think this is as expected - it's JRuby, too.

Reply:
>  the resolution on the other PR was just to declare that the behavior would remain incorrect on jruby for now

Please accept my apologies. I was wrong, and under the impression that the code behaves, and it's just that we can't write a test for JRuby because the Tempfile is implemented there differently.
But it surprised me that the code is affected, too.

Fixing this for other platforms, but not JRuby will certainly be an odd thing to do.

I'll need some time to have a closer look at this.

Reply:
No, I'm just failing to type correctly. I'm seeing that "JRuby 1.7 1.8 mode" has `RSpec::Support::Ruby.jruby?` NOT returning true; I tried to set up the two specs in question to skip on all jruby, and it worked for every container except this one :-\

> Please accept my apologies. I was wrong, and under the impression that the code behaves, and it's just that we can't write a test for JRuby because the Tempfile is implemented there differently.
> But it surprised me that the code is affected, too.

For what it's worth, the behavior jruby should stay broken in the same way that it is currently broken on all platforms. The change already merged in https://github.com/rspec/rspec-support/pull/598 actually fixes _that_ bug only on non-jruby platforms; I believe that's the reason.

Reply:
According to [this](https://github.com/rspec/rspec-support/pull/598#issuecomment-2200779633):

> both versions of Tempfile should behave the same from an API perspective

Can you please ask again, but with a code example where `clone` doesn’t work as expected?

Reply:
I called it a "bug" because it does appear to be intentional, but it's been under discussion as a problem in MRI a few times too; as far as I can tell, this is the only place in all of ruby that an object can change its type after instantiation. I would generally say that rubyists should avoid being concerned with the type of the object in the first place; that's just the only place the information I needed to check (is the the original tempfile, or has it been reopened?) seems to be held.

I.. could maybe patch File in rspec-support to track that explicitly; how offensive would that be?

Reply:
A wild idea - can we use a proxy that would additionally tell us if it’s a fake stream (and use a Tempfile underneath)?